src/include/stir/IO/ECAT8_32bitListmodeInputFileFormat.h

```diff
@@ -46,8 +46,22 @@ public InputFileFormat<ListModeData >
     actual_can_read(const FileSignature& signature,
 		    std::istream& input) const
   {
-    // TODO need to do check that it's a siemens list file etc
-    return is_interfile_signature(signature.get_signature());
+        std::string ecat8;
+
+    for (int i=0;i<signature.size(); i++)
+    {
+
+        ecat8=std::string(1,signature.get_signature()[i])+
+                std::string(1,signature.get_signature()[i+1])+
+                std::string(1,signature.get_signature()[i+2])+
+                std::string(1,signature.get_signature()[i+3])+
+                std::string(1,signature.get_signature()[i+4])+
+                std::string(1,signature.get_signature()[i+5])+
+                std::string(1,signature.get_signature()[i+6]);
```

Comment:
```suggestion
        const std::string ecat8(signature.get_signature() + i, 8);
```
should work I believe